tests/ops/functions/test_matrix.py

```diff
@@ -191,6 +191,51 @@ def testcircuit():
         assert np.allclose(matrix, expected_matrix)
 
 
+class TestWithParameterBroadcasting:
+    def test_multiple_operations_tape_single_broadcasted_op(self):
+        """Check the total matrix for a tape containing multiple gates
+        and a single broadcasted gate."""
+        wire_order = ["a", "b", "c"]
+
+        angles = np.array([0.0, np.pi, 0.0])
+        with qml.tape.QuantumTape() as tape:
+            qml.S(wires="b")

```

Comment:
Do we have a test for the case when the very first operation is broadcasted and there are non-broadcasted operations following?

Likely a variant to this and likely rearranging the order of ops here would yield that case already.

Reply:
Good idea, will include! :)